### PR TITLE
storage: instrument command queue waits and read path better

### PR DESCRIPTION
**Pull request by tbg**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9405)

### CODE REVIEW

File: storage/replica.go [hunk 1]
```diff
@@ -1153,15 +1153,20 @@ func (r *Replica) beginCmds(ctx context.Context, ba *roachpb.BatchRequest) (func
 		cmd = r.mu.cmdQ.add(readOnly, spans...)
 		r.mu.Unlock()
 
+		beforeWait := time.Now()
 		ctxDone := ctx.Done()
+		numChans := len(chans)
+		if numChans > 0 {
```
Comment (petermattis) [inline storage/replica.go]: Combine into `if numChans := len(chans); numChans > 0`?

**Review by petermattis — APPROVED** [review]
LGTM

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/replica.go, line 1159 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9405#-KRj8eclTOsuhDliE-1T-r1-1159:-KRj9Hi7MTdHKg4WntuQ:bt3j55o) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2b1f356db44828edf0b0348e12d188120f89fdd/storage/replica.go#L1159)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Combine into `if numChans := len(chans); numChans > 0`?
</details>
it's used below.

---

*[storage/replica.go, line 1429 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9405#-KRj9K-UUyXuXjM0Kofm:-KRj9K-UUyXuXjM0Kofn:bgeqfw2) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2b1f356db44828edf0b0348e12d188120f89fdd/storage/replica.go#L1429)):*
> ```Go
> 	}
> 
> 	log.Trace(ctx, "waiting for read lock")
> ```

no trace point after the wait?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9405#-:-KRj9OczRvmjSAodHo8b:bnfp4nl)*

File: storage/replica.go (hunk 1 above)
Comment (tschottdorf) [inline storage/replica.go]: I need `numChans` below the `for`.

**Comment by tschottdorf** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[storage/replica.go, line 1429 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9405#-KRj9K-UUyXuXjM0Kofm:-KRjAXuSCLVztBRqCxlY:b-65nzr6) ([raw file](https://github.com/cockroachdb/cockroach/blob/b2b1f356db44828edf0b0348e12d188120f89fdd/storage/replica.go#L1429)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> no trace point after the wait?
</details>
I'm not sure how to go about these things. We can't have two trace points for every little thing that happens (and the read lock in particular is uncontended except when splitting, so it won't be interesting). Will leave this one as is for now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9405)*

**Pull request merged**